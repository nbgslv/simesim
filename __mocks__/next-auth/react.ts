const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
