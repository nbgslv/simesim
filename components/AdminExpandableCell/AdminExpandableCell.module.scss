@import "styles/variables";

.button {
  color: $text-color !important;
}