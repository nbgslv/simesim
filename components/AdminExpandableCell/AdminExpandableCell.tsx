import React, { useEffect } from 'react';
import { Button } from 'react-bootstrap';
import styles from './AdminExpandableCell.module.scss';

const AdminExpandableCell = ({
  value,
  length = 50,
  shortValue,
}: {
  value: React.ReactNode | string;
  length?: number;
  shortValue?: string;
}) => {
  const [expanded, setExpanded] = React.useState(false);
  const [typedValue, setTypedValue] = React.useState<React.ReactNode | string>(
    null
  );

  useEffect(() => {
    if (typeof value === 'string') {
      setTypedValue(value);
    } else if (React.isValidElement(value)) {
      setTypedValue(React.cloneElement(value, { ...value.props }));
    }
  }, [value]);

  if (!typedValue) return null;

  return (
    <div>
      <span>
        {expanded
          ? typedValue
          : shortValue?.substring?.(0, length) ??
            (typedValue as string).substring(0, length)}
      </span>
      &nbsp;
      {(shortValue ?? (typedValue as string)).length > length && (
        <Button
          className={styles.button}
          variant="link"
          size="sm"
          onClick={() => setExpanded(!expanded)}
        >
          {expanded ? 'Hide' : 'Show more...'}
        </Button>
      )}
    </div>
  );
};

export default AdminExpandableCell;
