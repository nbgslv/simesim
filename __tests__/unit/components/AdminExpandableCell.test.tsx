import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import AdminExpandableCell from '../../../components/AdminExpandableCell/AdminExpandableCell';

describe('AdminExpandableCell', () => {
  const value =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non risus.';
  const shortValue = 'Lorem ipsum dolor';
  const length = 20;

  it('renders the short value by default', () => {
    render(
      <AdminExpandableCell
        value={value}
        shortValue={shortValue}
        length={length}
      />
    );
    expect(screen.getByText(shortValue)).toBeInTheDocument();
    expect(screen.queryByRole('button')).not.toBeInTheDocument();
  });

  it('expands and shows the full value when the button is clicked', () => {
    render(
      <AdminExpandableCell value={value} shortValue={value} length={length} />
    );
    expect(
      screen.getByText(value.substring(0, length)).textContent?.trim()
    ).toHaveLength(length);
    expect(screen.getByText(value.substring(0, length))).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'Show more...' }));
    expect(screen.getByText(value)).toBeInTheDocument();
  });

  it('shows the "Hide" button when the value is expanded', () => {
    render(
      <AdminExpandableCell value={value} shortValue={value} length={length} />
    );
    fireEvent.click(screen.getByRole('button', { name: 'Show more...' }));
    expect(screen.getByRole('button', { name: 'Hide' })).toBeInTheDocument();
  });

  it('hides the full value when the button is clicked again', () => {
    render(
      <AdminExpandableCell value={value} shortValue={value} length={length} />
    );
    fireEvent.click(screen.getByRole('button', { name: 'Show more...' }));
    fireEvent.click(screen.getByRole('button', { name: 'Hide' }));
    expect(
      screen.getByText(value.substring(0, length)).textContent?.trim()
    ).toHaveLength(length);
  });
});
