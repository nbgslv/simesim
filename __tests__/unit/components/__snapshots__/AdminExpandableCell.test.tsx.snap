// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminExpandableCell matches snapshot 1`] = `
<div>
  <div>
    <span>
      Lorem ipsum dolor
    </span>
    Â 
  </div>
</div>
`;
